bitnami/argo-cd/templates/applicationset/deployment.yaml

```
@@ -119,8 +119,7 @@ spec:
           livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.applicationSet.customLivenessProbe "context" $) | nindent 12 }}
           {{- else if .Values.applicationSet.livenessProbe.enabled }}
           livenessProbe:
-            httpGet:
-              path: /healthz
+            tcpSocket:
```

This would affect Kubescape score give this tool recommends using different probes for liveness and readiness checks. An alternative would be using sth such as "pgrep" to check the process is alive:

```suggestion
            exec:
              command:
                - pgrep
                - -f
                - argocd-applicationset-controller
```

Please give it a try.

> Note: I know this is stupid, but it's important to avoid the chart to obtain bad scores in these security scans.

---

I read some blogs about using different liveness and readiness probe. Everybody does seem to agree that they should be different, so I am fine with updating this. But just having it different for the sake of being different is indeed a little stupid. It is what it is. Hopefully ArgoCD will implement the health-check properly, and that would solve the issue!

Thanks for suggesting the `pgrep` command. I didn't know this existed! Tested it, and it worked as expected. I implemented this.